lib/public_key/src/pubkey_cert.erl

```diff
@@ -558,82 +635,578 @@ root_cert(Name, Opts) ->
+                                                                },
+                                                inhibit_any_policy = CurrentAnyPolicy} =
+                             ValidationState) ->
+    AnyPolicyConstraint = policy_constraint(CurrentAnyPolicy, InhibitAnyPolicy),
+    ValidationState#path_validation_state{inhibit_any_policy = AnyPolicyConstraint,
+                                          policy_inhibitany_ext = undefined};
+handle_inhibit_anypolicy(ValidationState) ->
+    ValidationState.
+
+policy_constraint(Current, asn1_NOVALUE) ->
+    Current;
+policy_constraint(Current, New) ->
+    erlang:min(Current, New).
+
+current_any_policy_qualifiers(Info) ->
+    case lists:keyfind(?anyPolicy, #'PolicyInformation'.policyIdentifier, Info) of
+        #'PolicyInformation'{policyQualifiers = AnyQualifiers} ->
+            AnyQualifiers;
+        _ ->
+            []
+    end.
+
+maybe_decrement(0, _) ->
+    0;
+maybe_decrement(N, false) ->
+    N-1;
+maybe_decrement(N, true) ->
+    N.
+
+%% End Prepare Next Cert Policy Handling ---------------------------------------
+
+%% Start Wrap Up Policy Handling RFC 5280 Section 6.1.5 %% ---------------------
+
+%% Step G from RFC
+
+policy_tree_intersection([?anyPolicy], Tree, _) -> % (ii) from RFC
+    Tree;
+policy_tree_intersection(UserPolicySet, Tree0, CurrentDepth) ->
+    case pubkey_policy_tree:is_empty(Tree0) of
+        true ->  % (i) from RFC
+            Tree0;
+        false -> % (iii) from RFC
+            ValidPolicyNodeSet = pubkey_policy_tree:valid_policy_node_set(Tree0),
+            InvalidNodes = apply_user_constraints(ValidPolicyNodeSet, UserPolicySet),
+            Tree1 = pubkey_policy_tree:prune_invalid_nodes(Tree0, InvalidNodes),
+            Tree = handle_any_policy_leaves(Tree1, ValidPolicyNodeSet, UserPolicySet, CurrentDepth),
+            pubkey_policy_tree:prune_tree(Tree)
+    end.
+
+apply_user_constraints(_, [?anyPolicy]) ->
+    [];
+apply_user_constraints(ValidPolicyNodeSet, UserPolicySet) ->
+    apply_user_constraints(ValidPolicyNodeSet, UserPolicySet, []).
+
+apply_user_constraints([], _, Acc) ->
+    Acc;
+apply_user_constraints([#{valid_policy := ?anyPolicy}| Rest],
+               UserPolicySet, Acc) ->
+    apply_user_constraints(Rest, UserPolicySet, Acc);
+apply_user_constraints([#{valid_policy := Policy} = Node | Rest],
+                UserPolicySet, Acc) ->
+    case lists:member(Policy, UserPolicySet) of
+        true ->
+            apply_user_constraints(Rest, UserPolicySet, Acc);
+        false ->
+            apply_user_constraints(Rest, UserPolicySet, [Node | Acc])
+    end.
+
+handle_any_policy_leaves(Tree, _, [?anyPolicy], _) ->
+    Tree;
+handle_any_policy_leaves(Tree0, ValidPolicyNodeSet, UserPolicySet, CurrentDepth) ->
+    case pubkey_policy_tree:any_leaf(Tree0, CurrentDepth) of
+        no_node ->
+            Tree0;
+        AnyLeaves ->
```

Comment:
- calling `any_leaf` (singular) and storing result in AnyLeaves (plural) - looks strange
- also `any_leaf` - in this particular case I think longer name and in plural form would be better, maybe `any_policy_leaves`

Reply:
@u3s check and resolve if OK

Reply:
it is plural consistently now